### PR TITLE
Bump elastic stack version automation

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Pipeline automation to bump the elastic stack version in the given list of repositories. This will create a PR if there are new changes to be applied.

## Why is it important?

No more issues when consuming latest snapshots.

## Related issues
Closes #ISSUE

## Tests

Manually tested with `dry-run` mode

image

image

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: .ci/bumpStackVersion.groovy [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def generateSteps(Map args = [:]) {
+  def projects = readYaml(file: '.ci/.bump-stack-version.yml')
+  def parallelTasks = [:]
+  projects['projects'].each { p ->
```
Comment (kuisathaverat) [inline .ci/bumpStackVersion.groovy:78]: an alternative implementation

```
matrix(
          agent: 'ubuntu-20',
          axes: [
            axis('JOB_PROJECT', projects ),
            axis('JOB_BRANCHES', branches )
          ]
        ){
        ...
        }
```

Comment (v1v) [reply .ci/bumpStackVersion.groovy:78]: Cool, I'll give a go!

Comment (v1v) [reply .ci/bumpStackVersion.groovy:78]: matrix does not work well with `Field` variables :/ 

```
10:22:33  ERROR: Pull Requests failed
10:22:33  Also:   groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  Also:   groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  groovy.lang.MissingPropertyException: No such property: latestVersions for class: groovy.lang.Binding
10:22:33  	at groovy.lang.Binding.getVariable(Binding.java:63)
10:22:33  	at org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SandboxInterceptor.onGetProperty(SandboxInterceptor.java:270)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker$7.call(Checker.java:353)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:357)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
10:22:33  	at org.kohsuke.groovy.sandbox.impl.Checker.checkedGetProperty(Checker.java:333)
10:22:33  	at com.cloudbees.groovy.cps.sandbox.SandboxInvoker.getProperty(SandboxInvoker.java:29)
10:22:33  	at com.cloudbees.groovy.cps.impl.PropertyAccessBlock.rawGet(PropertyAccessBlock.java:20)
10:22:33  	at WorkflowScript.generateSteps(WorkflowScript:60)
10:22:33  	at matrix.call(matrix.groovy:35)
```

So I'll not use it for the time being (9eacd58 is the commit with the proposal)

Comment (v1v) [reply .ci/bumpStackVersion.groovy:78]: I said nothing! I had a wrong variable name :/

image

**Review by cachedout — APPROVED** [review]

**Pull request merged**